src/DynamoPackages/PackageLoader.cs

```diff
@@ -373,12 +373,13 @@ public void LoadPackages(IEnumerable<Package> packages)
         /// <summary>
         /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
         /// </summary>
-        /// <param name="newPaths"></param>
-        /// <param name="preferences"></param>
+        /// <param name="packagePaths"></param>
+        /// <param name="nodePaths"></param>
         /// <param name="customNodeManager"></param>
-        public void LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreferences preferences, CustomNodeManager customNodeManager)
+        public void LoadCustomNodesAndPackages(IEnumerable<string> packagePaths, IEnumerable<string> nodePaths, 
```

Comment:
1. adding optional parameters is an API break resulting in a missing method exception: see: https://stackoverflow.com/questions/1456785/a-definitive-guide-to-api-breaking-changes-in-net / Adding a parameter with a default value.


2. changing parameter names on a public method is a source level break, code that uses named parameters will no longer compile.


please add a new method instead.

Reply:
This API was recently introduced to replace the other version of LoadCustomNodesAndPackages at line 432 which is now obsolete. However, this is not working for us quite as we expected and I believe we are the only ones that attempted to use it. Since there was no official release with this I thought we would simply be better off by changing it rather than creating another variant. Do you still think we should have a new method?

Reply:
No - you are correct @BogdanZavu, thanks for correcting me, I didn't see that @aparajit-pratap just added this method.

Reply:
New paths was meant to be specifically `new` paths. I'm not sure I'm okay with us changing this to be any path. ⚠️

Reply:
`nodePaths` and `packagePaths`? What's the difference? I can see how it can get confusing for API users/deveopers.

Reply:
unless you're referring to `nodePaths `as custom node paths only, in which case I'd suggest to name it as `customNodePaths` and document it as well.

Reply:
There was no enforcement whatsoever previously for being a "new" path and it was already used as any path.
Regarding the difference between packagePaths and nodePaths the difference is the same as before, nothing changed here, packages are loaded from a given location and custom nodes from another.
Regarding the naming, the utility is named LoadCustomNodesAndPackages which implies the custom nature of nodePaths parameter.

Anyway as an API user/developer I made this PR because the existing API is a little bit confusing and not working as one would expect. Also forcing an API developer to manipulate path manager to use this could have undesired effects if the changes to path manager are not reverted.

Reply:
If your assertion is true that this API "doesn't" work, then it would be best to leave this unchanged as an internal method and you create a new one for your purposes.

Reply:
There is no enforcement of new paths but it is described in the code comments and also implied by the name of the argument.